app/services/reachable_users_service.rb

```diff
@@ -110,4 +110,51 @@ def get_users_count
     query = ActiveRecord::Base.send(:sanitize_sql_array, [sql, params])
     ActiveRecord::Base.connection.execute(query).to_a.first
   end
+
+  def self.get_running_campaigns_for_user(user)
+    sql = <<~SQL.tr("\n", " ").squish
+      select
+      c.id as campaign_id,
+      c.available_doses,
+      COALESCE(cm.n_confirmed_matches, 0) as n_confirmed_matches,
+      COALESCE(um.n_user_matches, 0) as n_user_matches,
+      c.min_age,
+      c.max_age,
+      c.status,
+      v.name,
+      c.max_distance_in_meters/1000 as max_distance,
+      (SQRT(((v.lat - (:lat))*110.574)^2 + ((v.lon - (:lon))*111.320*COS((:lat)::float*3.14159/180))^2)) as distance
+      from campaigns c
+      left outer join vaccination_centers v on (v.id = c.vaccination_center_id)
```

Comment:
is there a reason to use `outer join` here?

Reply:
non, s'toi le boss